src/components/atoms/Plugin/hooks.ts

```diff
@@ -51,7 +52,8 @@ export const defaultIsMarshalable = (obj: any): boolean => {
     Array.isArray(obj) ||
     Object.getPrototypeOf(obj) === Function.prototype ||
     Object.getPrototypeOf(obj) === Object.prototype ||
-    obj instanceof Date
+    obj instanceof Date ||
+    typeof obj?.then === "function"
```

Comment:
Here i added `Promise` to be marshalable. I check `.then` function to check if it is a promise base on [this](https://stackoverflow.com/questions/27746304/how-to-check-if-an-object-is-a-promise). Please advice if there's any better way.

Reply:
Updated.